orbit/models/lgt.py

```diff
@@ -188,239 +193,145 @@ def _validate(regression_params, valid_length):
             raise ModelException(
-                "DataFrame does not contain specified regressor colummn(s)."
+                "DataFrame does not contain specified regressor column(s)."
             )
 
-        # validate response variable is in df
-        if self.response_col not in df_columns:
-            raise ModelException("DataFrame does not contain `response_col`: {}".format(self.response_col))
-
     def _set_regressor_matrix(self, df):
+        """Set regressor matrix based on the input data-frame.
+        Notes
+        -----
+        In case of absence of regression, they will be set to np.array with dim (num_of_obs, 0) to fit Stan requirement
+        """
         # init of regression matrix depends on length of response vector
         self._positive_regressor_matrix = np.zeros((self._num_of_observations, 0), dtype=np.double)
+        self._negative_regressor_matrix = np.zeros((self._num_of_observations, 0), dtype=np.double)
         self._regular_regressor_matrix = np.zeros((self._num_of_observations, 0), dtype=np.double)
 
         # update regression matrices
         if self._num_of_positive_regressors > 0:
             self._positive_regressor_matrix = df.filter(
-                items=self._positive_regressor_col,).values
+                items=self._positive_regressor_col, ).values
+
+        if self._num_of_negative_regressors > 0:
+            self._negative_regressor_matrix = df.filter(
+                items=self._negative_regressor_col, ).values
 
         if self._num_of_regular_regressors > 0:
             self._regular_regressor_matrix = df.filter(
-                items=self._regular_regressor_col,).values
-
-    def _set_init_values(self):
-        """Set Stan init as a callable
-
-        See: https://pystan.readthedocs.io/en/latest/api.htm
-        """
-        def init_values_function(seasonality):
-            init_values = dict()
-            if seasonality > 1:
-                seas_init = np.random.normal(loc=0, scale=0.05, size=seasonality - 1)
-                # catch cases with extreme values
-                seas_init[seas_init > 1.0] = 1.0
-                seas_init[seas_init < -1.0] = -1.0
-                init_values['init_sea'] = seas_init
+                items=self._regular_regressor_col, ).values
 
-            return init_values
-
-        seasonality = self._seasonality
-
-        # init_values_partial = partial(init_values_callable, seasonality=seasonality)
-        # partialfunc does not work when passed to PyStan because PyStan uses
-        # inspect.getargspec(func) which seems to raise an exception with keyword-only args
-        # caused by using partialfunc
-        # lambda as an alternative workaround
-        if seasonality > 1:
-            init_values_callable = lambda: init_values_function(seasonality)  # noqa
-            self._init_values = init_values_callable
+    @staticmethod
+    def _get_regressor_matrix(df, rr_col, pr_col, nr_col):
```

Comment:
I don't think this is used anywhere. If that is the case, let's remove to avoid confusion since we preserve regression order using `self._regressor_col`